### ISSUE TITLE
Propose to rename spacemacs-buffer.el to spacemacs-buffer.txt

### ISSUE THREAD

**Issue by zilongshanren** [syl20bnr/spacemacs#5958]
Since I enable semantic layer for emacs-lisp-mode, when Spacemacs starts up, it will load the spacemacs-buffer.el and it will also activate semantic and other minor mode for emacs-lisp-mode which is very time consuming...

So I have renamed the suffix of spacemacs-buffer.el to spacemacs-buffer.txt and it reduce about 1.5s for the loading time 😄

**Comment by zilongshanren** [comment]
![image](https://cloud.githubusercontent.com/assets/785541/14953334/55e8255e-109b-11e6-8776-d56d81e0b797.png)

**Comment by JAremko** [comment]
Impressive :shipit:

**Comment by bmag** [comment]
Wow, nice find, I had no idea. I assume by "renamed" you mean "changed the value of `spacemacs-buffer--cache-file`"?

Loading all these minor modes and other file-opening-related stuff is done because `spacemacs/dump-vars-to-file` uses `find-file-noselect` internally.

Rewriting `spacemacs/dump-vars-to-file` with `with-temp-file` should be simpler and avoid all the extra loading. I'm thinking something along the lines of:
```elisp
(with-temp-file filename
  (spacemacs/dump varlist buf)
  (make-directory (file-name-directory filename) t))
```
But feel free to see what works and make a PR. (not sure I'll have the time to do it myself before the weekend)

**Comment by zilongshanren** [comment]
@bmag 
I mean change the suffix name of `spacemacs-buffer.el` file to the `.txt` suffix, thus when loading `spacemacs-buffer.el` at startup, it will not activate many time consuming operations related to emacs-lisp-mode.

Or we could avoid loading `spacemacs-buffer.el` when spacemacs start up.

**Comment by bmag** [comment]
AFAICT `spacemacs-buffer.el` is a cache file and is accessed for two reasons: to read the cached value of `spacemacs-buffer--release-note-version`, and to write it back when it changes.
When writing the value to `spacemacs-buffer.el`, the current implementation causes Emacs to visit the file, although that isn't really necessary.
I can't test now, but I think the writing is done automatically at startup only if the release notes are displayed automatically at startup.

**Comment by bmag** [comment]
> Or we could avoid loading spacemacs-buffer.el when spacemacs start up.

The slowness is due to saving the file, not because of loading it. (surprisingly)

**Comment by zilongshanren** [comment]
@bmag  
When spacemacs starts up, it will load `~/.cache/bookmarks` and `~/.cache/spacemacs-buffer.el` file and when loading the spacemacs-buffer.el, it will cause performance lose...

**Comment by bmag** [comment]
The current implementation of `spacemacs/dump-vars-to-file`, located in `core-funcs.el`, is:
```elisp
(defun spacemacs/dump-vars-to-file (varlist filename)
  "simplistic dumping of variables in VARLIST to a file FILENAME"
  (save-excursion
    (let ((buf (find-file-noselect filename)))
      (set-buffer buf)
      (erase-buffer)
      (spacemacs/dump varlist buf)
      (make-directory (file-name-directory filename) t)
      (save-buffer)
      (kill-buffer))))
```

What I'm saying is that this code causes spacemacs-buffer.el to be loaded (technical term: "visited") at startup with `find-file-noselect`, but we can change it so it doesn't load the file. The file is also loaded (technical term: "loaded") elsewhere with `load`, but I doubt this is the problematic loading.

Can you rename spacemacs-buffer.txt back to spacemacs-buffer.el and replace the above code with:
```elisp
(defun spacemacs/dump-vars-to-file (varlist filename)
  "simplistic dumping of variables in VARLIST to a file FILENAME"
  (with-temp-file filename
    (spacemacs/dump varlist buf)
    (make-directory (file-name-directory filename) t)))
```
And tell me how the startup time compares to the previous results?
(I would have attached a diff file, but I'm currently on a window machine without a diff program)

**Comment by zilongshanren** [comment]
@bmag 
Sure, sir.

I got this error `Symbol's value as variable is void: buf`

**Comment by bmag** [comment]
oh, sorry, it should be `(current-buffer)` instead of `buf`:
```elisp
(defun spacemacs/dump-vars-to-file (varlist filename)
  "simplistic dumping of variables in VARLIST to a file FILENAME"
  (with-temp-file filename
    (spacemacs/dump varlist (current-buffer))
    (make-directory (file-name-directory filename) t)))
```

**Comment by zilongshanren** [comment]
@bmag 
Great! It works like my solution.
Thanks 👍

**Comment by d12frosted** [comment]
Fixed with released of Spacemacs v0.200. Let us know if you still have any problems with this issue.
